### ISSUE TITLE
Core dump running Python Lighting example on Raspberry Pi 4

### ISSUE THREAD

**Issue by sandeepmistry** [project-chip/connectedhomeip#21969]
#### Problem

I'm trying to run the [Python based lighting example](https://github.com/project-chip/connectedhomeip/tree/master/examples/lighting-app/python) on a Raspberry Pi 4 running Ubuntu 22.04 but get a core dump:

```
(python_env) root@ubuntu:/home/ubuntu/connectedhomeip-python/examples/lighting-app/python# python lighting.py 
[1660745048.570696][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_kvs-3HkuYN)
[1660745048.574383][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_kvs)
[1660745048.579531][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_factory.ini-HPFg6a)
[1660745048.579798][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_factory.ini)
[1660745048.580077][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_config.ini-8COg8H)
[1660745048.580246][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_config.ini)
[1660745048.580495][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_counters.ini-x0oe1E)
[1660745048.580656][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_counters.ini)
[1660745048.580986][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_factory.ini-Dhxgk3)
[1660745048.581580][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_factory.ini)
[1660745048.581647][79754:79754] CHIP:DL: NVS set: chip-factory/unique-id = "6E43A2356AA3FE56"
[1660745048.581842][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_factory.ini-C2Dly8)
[1660745048.582491][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_factory.ini)
[1660745048.582559][79754:79754] CHIP:DL: NVS set: chip-factory/vendor-id = 65521 (0xFFF1)
[1660745048.582744][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_factory.ini-IhvBPE)
[1660745048.583263][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_factory.ini)
[1660745048.583328][79754:79754] CHIP:DL: NVS set: chip-factory/product-id = 32769 (0x8001)
[1660745048.583542][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_counters.ini-T9ukQV)
[1660745048.584009][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_counters.ini)
[1660745048.584073][79754:79754] CHIP:DL: NVS set: chip-counters/reboot-count = 1 (0x1)
[1660745048.584246][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_counters.ini-qhIzBw)
[1660745048.584833][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_counters.ini)
[1660745048.584897][79754:79754] CHIP:DL: NVS set: chip-counters/total-operational-hours = 0 (0x0)
[1660745048.585075][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_counters.ini-BbysPT)
[1660745048.585869][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_counters.ini)
[1660745048.585931][79754:79754] CHIP:DL: NVS set: chip-counters/boot-reason = 0 (0x0)
[1660745048.586104][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_config.ini-iaRuzO)
[1660745048.586599][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_config.ini)
[1660745048.586659][79754:79754] CHIP:DL: NVS set: chip-config/regulatory-location = 0 (0x0)
[1660745048.586803][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_config.ini-KkBBCZ)
[1660745048.587847][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_config.ini)
[1660745048.587899][79754:79754] CHIP:DL: NVS set: chip-config/location-capability = 2 (0x2)
[1660745048.588413][79754:79754] CHIP:DL: Got Ethernet interface: eth0
[1660745048.588725][79754:79754] CHIP:DL: Found the primary Ethernet interface:eth0
[1660745048.589077][79754:79754] CHIP:DL: Got WiFi interface: wlan0
[1660745048.589143][79754:79754] CHIP:DL: Failed to reset WiFi statistic counts
[1660745048.589181][79754:79754] CHIP:DL: Device Configuration:
[1660745048.589255][79754:79754] CHIP:DL:   Serial Number: TEST_SN
[1660745048.589310][79754:79754] CHIP:DL:   Vendor Id: 65521 (0xFFF1)
[1660745048.589347][79754:79754] CHIP:DL:   Product Id: 32769 (0x8001)
[1660745048.589379][79754:79754] CHIP:DL:   Hardware Version: 0
Aborted (core dumped)
```

I had to apply the follow patch to get in building (will make a pull request for this at a later date):

```diff
diff --git a/scripts/build_python_device.sh b/scripts/build_python_device.sh
index 30779c6ab..894ea1bad 100755
--- a/scripts/build_python_device.sh
+++ b/scripts/build_python_device.sh
@@ -104,7 +104,7 @@ gn --root="$CHIP_ROOT" gen "$OUTPUT_ROOT" --args="chip_detail_logging=$chip_deta
 if [ "$enable_pybindings" == true ]; then
     ninja -v -C "$OUTPUT_ROOT" pycontroller
 else
-    ninja -v -C "$OUTPUT_ROOT" chip-library
+    ninja -v -C "$OUTPUT_ROOT" chip-core
 fi
 
 # Create a virtual environment that has access to the built python tools
@@ -115,7 +115,7 @@ virtualenv --clear "$ENVIRONMENT_ROOT"
 if [ "$enable_pybindings" == true ]; then
     WHEEL=("$OUTPUT_ROOT"/pybindings/pycontroller/pychip-*.whl)
 else
-    WHEEL=("$OUTPUT_ROOT"/controller/python/chip_library*.whl)
+    WHEEL=("$OUTPUT_ROOT"/controller/python/chip_core*.whl)
 fi
```

 #### Proposed Solution

N/A

**Comment by mrjerryjohns** [comment]
Thank you, a PR would be much apprecaited!

**Comment by sandeepmistry** [comment]
Some more insights on this after testing on a Ubuntu 20.04 x86 VM:

TLDR:

```
[1662659697.193593][16263:16263] CHIP:SPT: VerifyOrDie failure at ../../src/platform/CommissionableDataProvider.cpp:32: gCommissionableDataProvider != nullptr
```

Full app log:

```
$ python lighting.py 
[1662659697.185655][16263:16263] CHIP:DL: ChipLinuxStorage::Init: Using KVS config file: /tmp/chip_kvs
[1662659697.187538][16263:16263] CHIP:DL: writing settings to file (/tmp/chip_kvs-lhMDiX)
[1662659697.187868][16263:16263] CHIP:DL: renamed tmp file to file (/tmp/chip_kvs)
[1662659697.188096][16263:16263] CHIP:DL: ChipLinuxStorage::Init: Using KVS config file: /tmp/chip_factory.ini
[1662659697.188259][16263:16263] CHIP:DL: writing settings to file (/tmp/chip_factory.ini-pPjR6V)
[1662659697.188357][16263:16263] CHIP:DL: renamed tmp file to file (/tmp/chip_factory.ini)
[1662659697.188441][16263:16263] CHIP:DL: ChipLinuxStorage::Init: Using KVS config file: /tmp/chip_config.ini
[1662659697.188544][16263:16263] CHIP:DL: writing settings to file (/tmp/chip_config.ini-37RmoW)
[1662659697.188620][16263:16263] CHIP:DL: renamed tmp file to file (/tmp/chip_config.ini)
[1662659697.188702][16263:16263] CHIP:DL: ChipLinuxStorage::Init: Using KVS config file: /tmp/chip_counters.ini
[1662659697.188796][16263:16263] CHIP:DL: writing settings to file (/tmp/chip_counters.ini-sZPDgU)
[1662659697.188872][16263:16263] CHIP:DL: renamed tmp file to file (/tmp/chip_counters.ini)
[1662659697.189016][16263:16263] CHIP:DL: writing settings to file (/tmp/chip_factory.ini-CbDsgW)
[1662659697.189248][16263:16263] CHIP:DL: renamed tmp file to file (/tmp/chip_factory.ini)
[1662659697.189278][16263:16263] CHIP:DL: NVS set: chip-factory/unique-id = "0BCE0E82C21551DE"
[1662659697.189534][16263:16263] CHIP:DL: writing settings to file (/tmp/chip_factory.ini-deKb9U)
[1662659697.189732][16263:16263] CHIP:DL: renamed tmp file to file (/tmp/chip_factory.ini)
[1662659697.189818][16263:16263] CHIP:DL: NVS set: chip-factory/vendor-id = 65521 (0xFFF1)
[1662659697.189910][16263:16263] CHIP:DL: writing settings to file (/tmp/chip_factory.ini-sjk3xV)
[1662659697.190093][16263:16263] CHIP:DL: renamed tmp file to file (/tmp/chip_factory.ini)
[1662659697.190236][16263:16263] CHIP:DL: NVS set: chip-factory/product-id = 32769 (0x8001)
[1662659697.190329][16263:16263] CHIP:DL: writing settings to file (/tmp/chip_counters.ini-QtBUkW)
[1662659697.190652][16263:16263] CHIP:DL: renamed tmp file to file (/tmp/chip_counters.ini)
[1662659697.190760][16263:16263] CHIP:DL: NVS set: chip-counters/reboot-count = 1 (0x1)
[1662659697.190860][16263:16263] CHIP:DL: writing settings to file (/tmp/chip_counters.ini-lpFHPU)
[1662659697.191031][16263:16263] CHIP:DL: renamed tmp file to file (/tmp/chip_counters.ini)
[1662659697.191150][16263:16263] CHIP:DL: NVS set: chip-counters/total-operational-hours = 0 (0x0)
[1662659697.191246][16263:16263] CHIP:DL: writing settings to file (/tmp/chip_counters.ini-3Zty6W)
[1662659697.191433][16263:16263] CHIP:DL: renamed tmp file to file (/tmp/chip_counters.ini)
[1662659697.191518][16263:16263] CHIP:DL: NVS set: chip-counters/boot-reason = 0 (0x0)
[1662659697.191602][16263:16263] CHIP:DL: writing settings to file (/tmp/chip_config.ini-ZTtpLX)
[1662659697.191797][16263:16263] CHIP:DL: renamed tmp file to file (/tmp/chip_config.ini)
[1662659697.191915][16263:16263] CHIP:DL: NVS set: chip-config/regulatory-location = 0 (0x0)
[1662659697.192002][16263:16263] CHIP:DL: writing settings to file (/tmp/chip_config.ini-YOzlSV)
[1662659697.192215][16263:16263] CHIP:DL: renamed tmp file to file (/tmp/chip_config.ini)
[1662659697.192329][16263:16263] CHIP:DL: NVS set: chip-config/location-capability = 2 (0x2)
[1662659697.192530][16263:16263] CHIP:DL: Got Ethernet interface: enp0s5
[1662659697.192614][16263:16263] CHIP:DL: Found the primary Ethernet interface:enp0s5
[1662659697.192697][16263:16263] CHIP:DL: Failed to get WiFi interface
[1662659697.192718][16263:16263] CHIP:DL: Failed to reset WiFi statistic counts
[1662659697.192752][16263:16263] CHIP:DL: Device Configuration:
[1662659697.192799][16263:16263] CHIP:DL:   Serial Number: TEST_SN
[1662659697.193447][16263:16263] CHIP:DL:   Vendor Id: 65521 (0xFFF1)
[1662659697.193579][16263:16263] CHIP:DL:   Product Id: 32769 (0x8001)
[1662659697.193589][16263:16263] CHIP:DL:   Hardware Version: 0
[1662659697.193593][16263:16263] CHIP:SPT: VerifyOrDie failure at ../../src/platform/CommissionableDataProvider.cpp:32: gCommissionableDataProvider != nullptr
```
